spec/factories.rb

```diff
@@ -384,6 +385,9 @@
       payment_calculator = build(:calculator_per_item, preferred_amount: evaluator.payment_fee)
       payment_method = create(:payment_method, calculator: payment_calculator)
       create(:payment, order: order, amount: order.total, payment_method: payment_method, state: 'checkout')
+
+      # skip the rebuilding of order.shipments from line_items and stock locations (this is enforced in checkout step :address to :delivery)
+      order.stub(:create_proposed_shipments)
```

Comment:
but won't this leave the order in an unexpected state? why do we need to skip it?

Reply:
actually, the following method (that we are not skipping) is ensure_available_shipping_rates, and it passes.